README.md

```diff
@@ -33,132 +33,177 @@ Or install it yourself as:
+    # }
+    #
+    # `value` is exposed here, in case you want to get access to the argument's definition, including the type
+    # class which defines it, e.g. `metrics[:value].definition.metadata[:type_class]`
+    def argument_extracted(metrics)
+      store_metrics(:arguments, metrics)
+    end
+
+    private
+
+    def store_metrics(context_key, metrics)
+      ns = query.context.namespace(ANALYZER_NAMESPACE)
+      ns[:simple_extractor_results][context_key] ||= []
+      ns[:simple_extractor_results][context_key] << metrics
+    end
   end
+```
 
-  def field_extracted(metrics, _metadata)
-    Rails.logger.debug({
-      type_name: metrics[:type_name],           # "QueryRoot"
-      field_name: metrics[:field_name],         # "project"
-      deprecated: metrics[:deprecated],         # false
-      resolver_times: metrics[:resolver_times], # [0.1]
-    })
-  end
+Once defined, you can opt into capturing all metrics seen above by simply including GraphQL::Metrics as a plugin on your
+schema.
 
-  # NOTE: Applicable only if you set `use GraphQL::Batch, executor_class: GraphQLMetrics::TimedBatchExecutor`
-  # in your schema.
-  def batch_loaded_field_extracted(metrics, _metadata)
-    Rails.logger.debug({
-      key: metrics[:key],                                 # "CommentLoader/Comment"
-      identifiers: metrics[:identifiers],                 # "Comment/_/string/_/symbol/Class/?"
-      times: metrics[:times],                             # [0.1, 0.2, 4]
-      perform_queue_sizes: metrics[:perform_queue_sizes], # [3]
-    })
-  end
+### Make use of your analyzer
 
-  def argument_extracted(metrics, _metadata)
-    Rails.logger.debug({
-      name: metrics[:name],                           # "post"
-      type: metrics[:type],                           # "postInput"
-      value_is_null: metrics[:value_is_null],         # false
-      default_used: metrics[:default_used],           # false
-      parent_input_type: metrics[:parent_input_type], # "PostInput"
-      field_name: metrics[:field_name],               # "postCreate"
-      field_base_type: metrics[:field_base_type],     # "MutationRoot"
-    })
-  end
+Ensure that your schema is using the graphql-ruby 1.9+ `GraphQL::Execution::Interpreter` and `GraphQL::Analysis::AST`
+engine, and then simply add `use GraphQL::Metrics, analyzer: SimpleAnalyzer, options: { }`.
 
-  def variable_extracted(metrics, _metadata)
-    Rails.logger.debug({
-      operation_name: metrics[:operation_name],           # "MyMutation"
-      unwrapped_type_name: metrics[:unwrapped_type_name], # "PostInput"
-      type: metrics[:type],                               # "PostInput!"
-      default_value_type: metrics[:default_value_type],   # "IMPLICIT_NULL"
-      provided_value: metrics[:provided_value],           # false
-      default_used: metrics[:default_used],               # false
-      used_in_operation: metrics[:used_in_operation],     # true
-    })
-  end
+(TODO, what options should contain).
```

Comment:
Still TODO

Reply:
Yeah, I'll drop this any all mentions of `use GraphQL::Metrics` in this PR until it's unblocked in graphql-ruby. I'll leave a note for us in Shopify/shopify about it though.